admin/commands/management/remove.go

```diff
@@ -42,9 +45,53 @@ type removeMySQLCommand struct {
 	ServiceType string
 	ServiceName string
 	ServiceID   string
+	NodeID      string
 }
 
 func (cmd *removeMySQLCommand) Run() (commands.Result, error) {
+	if cmd.NodeID == "" {
+		status, err := agentlocal.GetStatus(agentlocal.DoNotRequestNetworkInfo)
+		if err != nil {
+			return nil, err
+		}
+		cmd.NodeID = status.NodeID
+	}
+
+	if cmd.ServiceID == "" && cmd.ServiceName == "" {
+		// PMM-10095: Automatic service lookup during removal
+		//
+		// Get services and remove it automatically once it's only one
+		// service registered
+		servicesRes, err := inventoryClient.Default.Services.ListServices(&services.ListServicesParams{
+			Body: services.ListServicesBody{
+				NodeID:      cmd.NodeID,
+				ServiceType: cmd.serviceType(),
+			},
+			Context: commands.Ctx,
+		})
+		if err != nil {
+			return nil, err
+		}
+		if len(servicesRes.Payload.Mysql) == 1 {
+			cmd.ServiceID = servicesRes.Payload.Mysql[0].ServiceID
+		}
+		if len(servicesRes.Payload.Mongodb) == 1 {
+			cmd.ServiceID = servicesRes.Payload.Mongodb[0].ServiceID
+		}
+		if len(servicesRes.Payload.Postgresql) == 1 {
+			cmd.ServiceID = servicesRes.Payload.Postgresql[0].ServiceID
+		}
+		if len(servicesRes.Payload.Proxysql) == 1 {
+			cmd.ServiceID = servicesRes.Payload.Proxysql[0].ServiceID
+		}
+		if len(servicesRes.Payload.Haproxy) == 1 {
+			cmd.ServiceID = servicesRes.Payload.Haproxy[0].ServiceID
+		}
+		if len(servicesRes.Payload.External) == 1 {
+			cmd.ServiceID = servicesRes.Payload.External[0].ServiceID
+		}
```

Comment:
I think we should make sure that we have only one service, with the current logic we might have 1 MySQL, 1 MongoDB, and 1 PostgreSQL service and it will automatically remove the PostgreSQL service.

Reply:
I don't think so because of filtering on request by res.serviceType(). That filters services

Reply:
oh, good point 👍🏼

Reply:
I think adding switch statement for type can make this part more obvious

Reply:
Yep. I rewrote the code